parser/main.py

```diff
@@ -188,215 +194,125 @@ def parse_request():
+    app.logger.info(f"Received a {message.type} message. ID = {id=}")
 
-    # try extracting measurements from CAN message
+    # try extracting measurements
     try:
-        extracted_measurements: List[Measurement] = can_msg.extract_measurements(CAR_DBC)
-        app.logger.info(f"Successfully parsed CAN message with id={can_msg.hex_identifier}({can_msg.identifier}) and placed into queue")
+        app.logger.info(f"Successfully parsed {type} message with id={id} and placed into queue")
     except Exception:
         app.logger.warn(
-            f"Unable to extract measurements for CAN message with id={can_msg.hex_identifier}({can_msg.identifier})")
+            f"Unable to extract measurements for {type} message with id={id}")
+        app.logger.warn(str(message.data["display_data"]))
         return {
             "result": "PARSE_FAIL",
-            "measurements": [],
-            "id": can_msg.identifier
+            "message": [],
+            "id": id
         }
 
-    # try putting the extracted measurements in the queue for Grafana streaming
-    try:
-        stream_queue.put(extracted_measurements, block=False)
-    except queue.Full:
-        app.logger.warn(
-            "Stream queue full. Unable to add measurements to stream queue!"
-        )
-
     # try writing the measurements extracted
-    for measurement in extracted_measurements:
-        name = measurement.name
-        source = measurement.source
-        m_class = measurement.m_class
-        value = measurement.value
+    for i in range(len(message.data[list(message.data.keys())[0]])):
+        name = message.data["Measurement"][i]
+        source = message.data["Source"][i]
+        m_class = message.data["Class"][i]
+        value = message.data["Value"][i]
+
 
         point = influxdb_client.Point(source).tag("car", CAR_NAME).tag(
             "class", m_class).field(name, value)
-
+        
         # write to InfluxDB
         try:
-            write_api.write(bucket=INFLUX_PROD_BUCKET, org=INFLUX_ORG, record=point)
+            write_api.write(bucket=message.type + bucket, org=INFLUX_ORG, record=point)
             app.logger.info(
-                f"Wrote '{name}' measurement to url={INFLUX_URL}, org={INFLUX_ORG}, bucket={INFLUX_PROD_BUCKET}!")
-        except Exception:
+                f"Wrote '{name}' measurement to url={INFLUX_URL}, org={INFLUX_ORG}, bucket={INFLUX_DEBUG_BUCKET}!")
+        except Exception as e:
             app.logger.warning("Unable to write measurement to InfluxDB!")
             return {
                 "result": "INFLUX_WRITE_FAIL",
-                "measurements": extracted_measurements,
-                "id": can_msg.identifier
+                "message": str(e),
+                "id": id,
+                "type": type
             }
 
     return {
         "result": "OK",
-        "measurements": extracted_measurements,
-        "id": can_msg.identifier
+        "message": message.data["display_data"],
+        "id": id,
```

Comment:
I don't think the ID should be included in the response. ID is only meaningful for the v2.0.0 CAN implementation. Instead, you'd have whatever data is in the "ID" field as another value associated with the "display_fields" key. So for CAN, you'd have that hex ID. For GPS, you'd have the timestamp as a "Timestamp" field (which it looks like you already have, so it's redundant there), and so on.